server/game/cards/08.5-TFM/SecretPact.js

```diff
@@ -0,0 +1,17 @@
+const DrawCard = require('../../drawcard.js');
+
+class SecretPact extends DrawCard {
+    setupCardAbilities(ability) {
+        this.attachmentRestriction(card => card.getType() === 'character' && !card.isFaction('martell'));
+        this.whileAttached({
+            effect: [
+                ability.effects.cannotBeDeclaredAsAttacker(), // TODO need generalization for melee
```

Comment:
This needs to be split off into a separate ```whileAttached``` with a ```condition``` property as it's only for challenges initiated against you. This card is interesting as it can be used as both a positive and a negative attachment.

Reply:
So, this was the point of my TODO comment. To elaborate: in a joust setting, one can only initiate challenges against its (sole) opponent (as they cannot initiate challenges against themselves), so AFAICT the implementation is *currently* correct. It will indeed break when we implement melee, hence my comment.
(I'm fine with requiring that new code should be melee-friendly, I'm commenting here just in case there is a simpler misunderstanding here.)

Reply:
As it currently stands, attaching this to a character you control yourself, makes the character unable to be declared as an attacker in any case. Which is not correct as the parent character should only be unable to be declared as an attacker in a challenge against yourself. So attaching it to a character you control basically makes that whole drawback go away.

Splitting it off into a separate ```whileAttached``` with the following condition:

```javascript
condition: () => this.game.currentChallenge && this.game.currentChallenge.defendingPlayer === this.controller
```

should fix this for both Joust and Melee purposes.

Reply:
You're absolutely right, my bad (I was focusing on the more complex melee case, and overlooked the drawback of being unable to attack when the attachment is used "positively"). Fixed.